railties/lib/rails/application.rb

```diff
@@ -511,6 +515,11 @@ def validate_secret_key_config! #:nodoc:
 
         if secrets.secret_token.blank?
           raise "Missing `secret_key_base` for '#{Rails.env}' environment, set this value in `config/secrets.yml`"
+        else
+          ActiveSupport::Deprecation.warn(<<-MSG.squish)
+           secrets.secret_token is deprecated and will be removed in Rails 6.0.
+           Please use secrets.secret_key_base instead.
+          MSG
```

Comment:
Let's unify these 2 deprecation warnings into here and wrap it in `secrets.secret_token.present?`: https://github.com/rails/rails/blob/db844d9fb30cd34411fcdcc6780f82aaa5471e81/railties/lib/rails/application.rb#L395